### PR TITLE
adding file:path model

### PR DESCRIPTION
**Pull request by vertexmc**

### CODE REVIEW

File: synapse/models/files.py [hunk 1]
```diff
@@ -83,3 +95,50 @@ def seedFileGoodHash(prop,valu,**props):
 
 #def revDataModel(core):
 
+class FileBaseType(DataType):
+
+    def norm(self, valu, oldval=None):
+
+        if not (s_compat.isstr(valu) and not valu.find('/') > -1 and len(valu) > 0):
```
Comment (vertexmc) [inline synapse/models/files.py]: Filename can't contain `/` or be empty string.

File: synapse/models/files.py [hunk 2]
```diff
@@ -83,3 +95,50 @@ def seedFileGoodHash(prop,valu,**props):
...
+
+        valu = valu.replace('\\', '/').lower().strip('/')
+        parts = valu.split('/')
+
+        props = {}
+        base = parts[-1]
+        if base:
+            props['base'] = base
+
+        dirname = '/'.join(parts[0:-1]).strip('/')
```
Comment (invisig0th) [inline synapse/models/files.py]: the last strip call here is redundant

File: synapse/tests/test_datamodel.py [hunk 3]
```diff
@@ -284,3 +284,62 @@ def test_datamodel_typefns(self):
...
+            self.assertRaises(s_datamodel.BadTypeValu, model.getTypeNorm, prop, bad)
+            self.assertRaises(s_datamodel.BadTypeValu, model.getTypeParse, prop, bad)
+            self.assertEqual((None, {}), model.getTypeFrob(prop, bad))
+
+    def test_datamodel_filebase(self):
+        model = s_datamodel.DataModel()
+        prop = 'file:base'
+
+        data = (
+            ('my_COOL_file', ('my_cool_file',{}), 'my_COOL_file'),
```
Comment (invisig0th) [inline synapse/tests/test_datamodel.py]: great test cases!

File: synapse/models/files.py (hunk 2 above)
Comment (vertexmc) [inline synapse/models/files.py]: thanks! removed

**Pull request merged**